highlight/highlight
# 4914 slack caching strategy

## Summary
Addresses https://github.com/highlight/highlight/issues/4914

Removes the manual  "sync" button and instead makes a throttled request for the latest
and greatest slack data. **Note, the slack API rate limits at ~20 requests/min so we throttle to respect that**

## How did you test this change?

1. navigate to a menu where you can select slack channels or `@` mentions users
2. type in the name of a channel that doesn't exist/shows no results
3. see the loading indicator in the search select menu
4. create the channel in your slack instance
5. search again
6. see it appear after the throttled request completes
7. rejoice 🎉  

<img width="518" alt="Screenshot 2023-04-18 at 11 29 05 PM" src="https://user-images.githubusercontent.com/1341253/232967266-6830a22d-0fd9-4e0e-a909-a747a1588eda.png">


## Are there any deployment considerations?
This will add more server traffic for storing slack info in postgres. Probably nothing noticeable though given that users aren't doing this all the time.
